batchop/env.py

```diff
@@ -64,9 +66,20 @@ def run_migrations_online() -> None:
         poolclass=pool.NullPool,
     )
 
+    @event.listens_for(connectable.engine, "connect")
+    def do_connect(dbapi_connection, correction_record):
+        # disable pysqlite's emitting of the BEGIN statement entirely.
+        # also stops it from emitting COMMIT before any DDL.
+        dbapi_connection.isolation_level = None
+
+    @event.listens_for(connectable.engine, "begin")
+    def do_begin(conn):
+        # emit our own BEGIN
+        conn.exec_driver_sql("BEGIN")
```

Comment:
Not sure why, but this actually looks fixes an issue.

```
$ ./reproduce_script.sh
+ rm alembic.db
+ alembic upgrade 90823bd3809e
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 90823bd3809e, create foo table
+ echo .tables
+ sqlite3 alembic.db
alembic_version  foo
+ echo '.schema foo'
+ sqlite3 alembic.db
CREATE TABLE foo (
	foo_id INTEGER NOT NULL,
	bar VARCHAR(512) NOT NULL,
	PRIMARY KEY (foo_id)
);
+ alembic upgrade head
+ true
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 90823bd3809e -> 8e32f2af1bd2, Add schema & data migration
Traceback (most recent call last):
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/bin/alembic", line 8, in <module>
    sys.exit(main())
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/config.py", line 590, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/config.py", line 584, in main
    self.run_cmd(cfg, options)
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/config.py", line 561, in run_cmd
    fn(
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/command.py", line 322, in upgrade
    script.run_env()
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/script/base.py", line 569, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 94, in load_python_file
    module = load_module_py(module_id, path)
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 110, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/c-bata/sandbox/alembic-sandbox/batchop/env.py", line 92, in <module>
    run_migrations_online()
  File "/Users/c-bata/sandbox/alembic-sandbox/batchop/env.py", line 86, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/runtime/environment.py", line 853, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/Users/c-bata/sandbox/alembic-sandbox/venv/lib/python3.10/site-packages/alembic/runtime/migration.py", line 623, in run_migrations
    step.migration_fn(**kw)
  File "/Users/c-bata/sandbox/alembic-sandbox/batchop/versions/8e32f2af1bd2_add_schema_data_migration.py", line 37, in upgrade
    raise Exception("An exception is raised during the data migration")
Exception: An exception is raised during the data migration
+ echo .tables
+ sqlite3 alembic.db
alembic_version  foo
+ echo '.schema foo'
+ sqlite3 alembic.db
CREATE TABLE foo (
	foo_id INTEGER NOT NULL,
	bar VARCHAR(512) NOT NULL,
	PRIMARY KEY (foo_id)
);
+ echo 'select * from foo;'
+ sqlite3 alembic.db
1|data 1
2|data 2
3|data 3
```

Reply:
Here is a diff of  `sqlalchemy.engine` info logs.

<img width="1496" alt="Screenshot 2022-09-29 11 45 35" src="https://user-images.githubusercontent.com/5564044/192927239-d63f7b8c-db24-41b0-8520-2b6898b05b12.png">

Reply:
> What’s “BEGIN (implicit)”?
> You might have noticed the log line “BEGIN (implicit)” at the start of a transaction block. “implicit” here means that SQLAlchemy did not actually send any command to the database; it just considers this to be the start of the DBAPI’s implicit transaction. You can register [event hooks](https://docs.sqlalchemy.org/en/14/core/events.html#core-sql-events) to intercept this event, for example.
> https://docs.sqlalchemy.org/en/14/tutorial/dbapi_transactions.html#:~:text=What's%20%E2%80%9CBEGIN%20(implicit)%E2%80%9D,of%20the%20DBAPI's%20implicit%20transaction.